cocoa/src/toga_cocoa/icons.py

```diff
@@ -18,11 +18,14 @@ def __init__(self, interface, path):
             # as an indicator that this is the app's default icon.
             # This bundle icon file definition might not contain an extension,
             # even thought the actual file will; so force the .icns extension.
-            bundle_icon = Path(
-                NSBundle.mainBundle.objectForInfoDictionaryKey("CFBundleIconFile")
+            bundle_icon = NSBundle.mainBundle.objectForInfoDictionaryKey(
+                "CFBundleIconFile"
             )
+            if bundle_icon is None:
+                # Not an .app bundle (e.g., POSIX build made with PyInstaller)
```

Comment:
```suggestion
                # Not an .app bundle (e.g., POSIX build made with PyInstaller)
                # This can't be tested as the testbed app is bundled by Briefcase.
```